test/test_misc.py

```diff
@@ -341,6 +341,31 @@ def test_memoize_on_disk_with_pym_expr():
     assert cached_result == uncached_result
 
 
+def test_basicset_keybuilder():
+    # See https://github.com/inducer/loopy/issues/912 for context
+    import islpy as isl
+
+    # Both sets have the same variables and constraints, but in different order.
+    # These sets are generated in test_convolution() in test_apps.py
+    a = isl.BasicSet("[im_w, im_h, nimgs, nfeats] -> "
+                     "{  : im_w >= 7 and im_h >= 7 and nimgs >= 0 and nfeats > 0 }")
+
+    b = isl.BasicSet("[nfeats, nimgs, im_h, im_w] -> "
+                     "{  : nfeats > 0 and nimgs >= 0 and im_h >= 7 and im_w >= 7 }")
+
+    from loopy.tools import LoopyKeyBuilder
+
+    # Equality
+    assert a == b
+    assert a.is_equal(b)
+    assert not a.plain_is_equal(b)
+
+    # Hashing
+    assert hash(a) != hash(b)
+    assert a.get_hash() != b.get_hash()
+    assert LoopyKeyBuilder()(a) == LoopyKeyBuilder()(b)
```

Comment:
This behavior is pretty :hankey:-tastic. I'm not super interested in codifying it with a test.

If I could snap my finger and have a solution enacted globally, probably I would like `==`, `plain_is_equal` and `__hash__` to all share a notion of equality. Something for namedisl to aspire to.

Reply:
I think `namedisl` is the only reasonable path forward. I don't know that I'd like to mess with the status quo too much until that time, unless there is a pressing reason. Is there one, in the context of this PR?

Reply:
I don't think there is a pressing need for this PR. The goal was to enable running the tests with LOOPY_ABORT_ON_CACHE_MISS (https://github.com/inducer/loopy/pull/828).